plugins/tech-insights-backend/src/service/router.ts

```diff
@@ -132,7 +132,15 @@ export async function createRouter<
   router.get('/facts/latest', async (req, res) => {
     const { entity } = req.query;
     const { namespace, kind, name } = parseEntityRef(entity as string);
-    const ids = req.query.ids as string[];
+
+    if (!req.query.ids) {
+      return res
+        .status(422)
+        .send({ error: 'Failed to parse ids from request' });
+    }
+    const ids = Array.isArray(req.query.ids)
+      ? (req.query.ids as string[])
+      : ([req.query.ids] as string[]);
```

Comment:
a neat consistent way is `const ids = [req.query.ids].flat()`

Reply:
Ah, that looks much better. Swapped into that.